### ISSUE TITLE
Behavior of reset rendering pool in development mode

### ISSUE THREAD

**Issue by eliias** [shakacode/react_on_rails#969]
Hello, I am using SSR along with styled-components and HMR and it is working quite well. But I run into a nasty problem today that at least needs some discussion. I have a small issue with the code that resets the rendering pool:

`gems/react_on_rails-10.0.0/lib/react_on_rails/server_rendering_pool/exec.rb`

```
if Utils.using_webpacker? && Webpacker.dev_server.running?
          return if @last_loaded_server_bundle == server_bundle_js_file_path
          @last_loaded_server_bundle = server_bundle_js_file_path
else
          # we're not hashing the server name and we can use the mtime
          file_mtime = File.mtime(server_bundle_js_file_path)
          @server_bundle_timestamp ||= file_mtime
          return if @server_bundle_timestamp == file_mtime
          @server_bundle_timestamp = file_mtime
end
```

The check in line two, just compares the path of the current server bundle file with a potential new one. As long as webpacker and dev_server are runnnig and if server bundle does not change its name, any changes in code will not get reflected in SSR output. This leads to potential issues in development. The markup is not updated on initial load and also (if you use custom keys in the generator function) potential additional output (e.g. styled-components SSR extracts CSS into style tags). My concrete problem is with prop name mismatches on rehydration, cause the server bundle does not reflect the changes.

Have you considered to use `mtime` check only and remove the optimization or did I miss something here? I guess in most projects this is not an issue cause HMR replaces the module and it will not result in prop name conflicts.

**Comment by justin808** [comment]
@eliias We're pretty slammed right now. Any chance that you'd be interested in our http://www.shakacode.com/work/shakacode-pro-support.pdf program? If so, we'll expedite the handling of your request. Otherwise, please research and potentially send us a PR.

**Comment by justin808** [comment]
@eliias this seems like a bug, as you've pointed out as your server file might not be using a filename that has a hash.

How about if we just always use the timestamp (else part)? I don't see any big issue with checking the mtime of the file.

```ruby
          # we're not hashing the server name and we can use the mtime
          file_mtime = File.mtime(server_bundle_js_file_path)
          @server_bundle_timestamp ||= file_mtime
          return if @server_bundle_timestamp == file_mtime
          @server_bundle_timestamp = file_mtime
```

**Comment by justin808** [comment]
Fixed in #1059
